File: ABS/C/ABC086C.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int N,t,x,y,i,tt,tx,ty;
    scanf("%d", &N);
    t=0; x=0; y=0;

    for(i=0; i<N; i++) {
        scanf("%d %d %d", &tt,&tx,&ty);
        int l=abs(tx-x)+abs(ty-y);
        if((l-(tt-t))>0 || (tt-t-l)%2)
            break;
        t = tt;
        x = tx;
        y = ty;
    }

    printf((i==N) ? "Yes\n" : "No\n");
}

File: ABS/Ruby/ABC086C.rb
n = gets.to_i
data = n.times.map { gets.strip.split.map(&:to_i) }

def is_reachable(crn, data)
  tc, xc, yc = crn
  tn, xn, yn = data
  d = (xn-xc).abs+(yn-yc).abs
  t = tn-tc
  return false if d>t
  (d-t).even?
end

crn = [0, 0, 0]
data.each do |d|
  unless is_reachable(crn, d)
    puts 'No'
    exit 0
  end
  crn = d
end

puts 'Yes'
